### ISSUE TITLE
Dev server doesn’t respond to ESBuild service crashing/being killed

### ISSUE THREAD

**Issue by huw** [remix-run/remix#6782]
### What version of Remix are you using?

1.18.1

### Are all your remix dependencies & dev-dependencies using the same version?

- [X] Yes

### Steps to Reproduce

1. Use the v2 dev server
2. Kill the ESBuild process (fastest way is to sort `htop` by RAM usage)
3. Edit a watched file

The context in which I am regularly seeing this is that my machine (GitHub Codespaces) only has 8GB of RAM and I am editing a medium-sized project. This causes my OS to evict the ESBuild server, which is using ~2GB of RAM (BTW—is this normal?)

### Expected Behavior

Either the dev server should restart the ESBuild service, or the server should gracefully crash.

### Actual Behavior

I get:

```
 info  rebuilding... (~ src/app/root.tsx)
The service is no longer running
```

Editing the file again produces the same message, ad infinitum. There is no way to recover the ESBuild server without restarting the Remix dev server.

**Comment by pcattori** [comment]
Closed by #7158

**Comment by huw** [comment]
(FWIW @pcattori I think there were also some RAM-saving improvements that you & the others shipped lately that helped here, since I haven’t seen it in a while—thank you!)
